backend/src/index.ts

```diff
@@ -0,0 +1,128 @@
+import { ApolloServer } from "@apollo/server";
+import { expressMiddleware } from "@apollo/server/express4";
+import { ApolloServerPluginDrainHttpServer } from "@apollo/server/plugin/drainHttpServer";
+import express from "express";
+import http from "http";
+import cors from "cors";
+import { WebSocketServer } from "ws";
+import { useServer } from "graphql-ws/lib/use/ws";
+import { makeExecutableSchema } from "@graphql-tools/schema"; 
+import { typeDefs } from "./schema/index";
+import { resolvers } from "./schema/index";
+import cookieParser from "cookie-parser";
+import dbConnect from "./config/dbConnection";
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+import User from "./models/User";
+import { MyContext } from "./schema/types/types";
+import { DecodedUser } from "./schema/types/types";
+import { PubSub } from "graphql-subscriptions"
+
+dotenv.config();
+
+// Create an Express app and HTTP server
+const app = express();
+const httpServer = http.createServer(app);
+
+// Create a GraphQL schema (needed for subscriptions)
+const schema = makeExecutableSchema({ typeDefs, resolvers });
+
+// Create WebSocket server
+const wsServer = new WebSocketServer({
+  server: httpServer,
+  path: "/graphql",
+});
+
+// Set up GraphQL WebSocket server
+const wsServerCleanup = useServer(
+  {
+    schema,
+    context: async (ctx:any) => {
+      // Extract user from connectionParams in WebSocket requests
+      let user: DecodedUser | null = null;
+      const token = ctx.connectionParams?.token;
+
+      if (token) {
+        try {
+          const decoded = jwt.verify(token, process.env.KEY as string);
+          if (typeof decoded === "object") {
+            const existingUser = await User.findById(decoded.id);
+            if (!existingUser) return { user: null };
+            user = decoded as DecodedUser;
+          }
+        } catch (error) {
+          console.error("Invalid WebSocket Token");
+        }
+      }
+
+      return { user };
```

Comment:
unnecessary declared variable user with let.
try to avoid use of let.
code could be more optimised like : 
```suggestion
      const token = ctx.connectionParams?.token;
      
      if (token) {
        try {
          const user: DecodedUser = jwt.verify(token, process.env.KEY as string);
          
          if (typeof decoded === "object") {
            const existingUser = await User.findById(decoded.id);
            
            if (!existingUser) return null;
            
            return user;
          }
        } catch (error) {
          console.error("Invalid WebSocket Token");
          return error;
        }
      }
```